### ISSUE TITLE
repos update and repos update all fail with error

### ISSUE THREAD

**Issue by redundancydisorder** [errbotio/errbot#1409]
In order to let us help you better, please fill out the following fields as best you can:

### I am...

* [X] Reporting a bug
* [ ] Suggesting a new feature
* [ ] Requesting help with running my bot
* [ ] Requesting help writing plugins
* [ ] Here about something else

### I am running...

* Errbot version: 6.1.2
* OS version: Ubuntu 18.04
* Python version: 3.6.9
* Using a virtual environment: no

### Issue description

Error when !repos update or !repos update all commands are run

Error in log:
```
/errbot/core_plugins/plugins.py", line 128, in repos_update
    for plugin in self._bot.plugin_manager.getAllPlugins():
AttributeError: 'BotPluginManager' object has no attribute 'getAllPlugins'
```

Bot also returns 
`'BotPluginManager' object has no attribute 'getAllPlugins'`

### Steps to reproduce

Install plugin via `!repos install` either url or name from list/search 
Run `!repos update all`

### Additional info

Closed issue https://github.com/errbotio/errbot/pull/1256 has reference to getAllPlugins But code was rejected due to other structural issues. getAllPlugins has been removed from all other places in the bot code in various commits and updates over the couple years.

**Comment by redundancydisorder** [comment]
Fix for getAllPlugins put together. But this then discovers another deeper issue error. 

repos_update  uses repo_manager.update_repos which does git_pull. Which is in util.py (the root one) which does a porcelain.pull(repo_path) Which fails for two reasons. 

First we are only providing the arg repo_path and not the arg remote_location. This errors out with "looking up remote from branch config not supported yet" 

Second. The initial git_clone (from utils.py) used when the plugin is installed does not configure the local repo confg correctly to enable a git pull even from the command line.  Error: "There is no tracking information for the current branch." Which can be fixed on the command line with: 
 `git branch --set-upstream-to=origin/master master`

I am looking to find a solution to both. If we can store the url or reconstruct it we can address the first issue. The second issue may just be adding some parameters or args to git_clone.

porcelain is from https://github.com/dulwich/dulwich

**Comment by sijis** [comment]
PR #1410 addressed the original issue.

Setting the tracking for a repo should also be fixed.

**Comment by sijis** [comment]
PR #1413  will try to address the ` "There is no tracking information for the current branch."` message.

**Comment by sijis** [comment]
This fix should be in 6.1.4 and greater.
